### PR TITLE
RavenDB-22957 Save server certificate with RW group permissions

### PR DESCRIPTION
**Pull request by hiddenshadow21**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22957

### Additional description

Set RW group permissions for server certificate when using Wizard Setup on Posix System.

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [x] Yes. Please list the affected platforms.
    - Posix systems
- [ ] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Voron/Platform/Posix/PosixHelper.cs [hunk 1]
```diff
@@ -129,5 +130,11 @@ public static void EnsurePathExists(string file)
             }
             dirsToCreate.ForEach(x => Directory.CreateDirectory(x));
         }
+
+        public static void EnsureRWPermissionsForOwnerAndGroup(string filePath)
+        {
+            var fileInfo = new UnixFileInfo(filePath);
+            fileInfo.FileAccessPermissions = fileInfo.FileAccessPermissions | FileAccessPermissions.UserRead | FileAccessPermissions.UserWrite | FileAccessPermissions.GroupRead | FileAccessPermissions.GroupWrite;
```
Comment (gregolsky) [inline src/Voron/Platform/Posix/PosixHelper.cs]: @ppekrol is that how we do it in RavenDB? Mono.Unix namespace?

Comment (gregolsky) [reply src/Voron/Platform/Posix/PosixHelper.cs]: Try https://learn.microsoft.com/en-us/dotnet/api/system.io.file.setunixfilemode?view=net-8.0

Comment (hiddenshadow21) [reply src/Voron/Platform/Posix/PosixHelper.cs]: It was introduced in .Net 7. IIRC we have still do a custom release of v5.4 on .Net 6.

Comment (ppekrol) [reply src/Voron/Platform/Posix/PosixHelper.cs]: Use .NET impl, I will deal with .NET 6 if needed

**Review by ppekrol — APPROVED** [review]
Approved, but please fix the warning

**Review by gregolsky — APPROVED** [review]

**Pull request merged**